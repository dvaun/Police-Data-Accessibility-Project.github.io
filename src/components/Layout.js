import React from "react"
import Header from "../components/Header"
import "../css/main.css"
import { StaticQuery, graphql } from "gatsby"
import { Helmet } from "react-helmet"

export default function Layout(props) {
  return (
    <StaticQuery
      query={graphql`
        query siteMetadata {
          site {
            siteMetadata {
              title
            }
          }
        }
      `}
      render={data => (
        <div>
          <Helmet>
            <meta charSet="utf-8" />
            <title>{data.site.siteMetadata.title}</title>
          </Helmet>
          <Header pages={props.pages} />
          <main className="container">{props.children}</main>
        </div>
      )}
    />
  )
}

Layout.defaultProps = {
  pages: [],
  title: "Title",
  children: null,
}
