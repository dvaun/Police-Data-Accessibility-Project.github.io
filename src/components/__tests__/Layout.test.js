import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"
import Layout from "../Layout"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Default Starter`,
        },
      },
    })
  )
})

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Layout />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
