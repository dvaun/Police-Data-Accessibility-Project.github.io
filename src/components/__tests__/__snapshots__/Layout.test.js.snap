// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <header
    className="header"
  >
    <a
      className="header__logo"
      href="/"
    >
      PDAP
    </a>
    <nav
      className="nav"
    >
      <ul
        className="nav__menu"
      />
    </nav>
  </header>
  <main
    className="container"
  />
</div>
`;
